kythe/web/site/examples.md

```diff
@@ -113,6 +113,90 @@ $ /opt/kythe/tools/runextractor cmake \
     -sourcedir=$CMAKE_ROOT_DIRECTORY
 ```
 
+## Extracting Gradle based repositories
+
+1. Install compiler wrapper
+
+Extraction works by intercepting all calls to `javac` and saving the compiler arguments and inputs to a "compilation unit", which is stored in a .kzip file. We have a javac-wrapper.sh script that forwards javac calls to the java extractor and then calls javac. Add this to the end of your project's build.gradle:
+
+```groovy
+allprojects {
+  gradle.projectsEvaluated {
+    tasks.withType(JavaCompile) {
+      options.fork = true
+      options.forkOptions.executable = '/opt/kythe/extractors/javac-wrapper.sh'
+    }
+  }
+}
+```
+
+2. VName configuration
+
+Next, you will need to create a vnames.json mapping file, which tells the
+extractor how to assign vnames to files based on their paths. A basic vnames
+config for a gradle project looks like:
+
+```json
+// vnames.json
+[
+  {
+    "pattern": "(build/[^/]+)/(.*)",
+    "vname": {
+      "corpus": "MY_CORPUS",
+      "path": "@2@",
+      "root": "@1@"
+    }
+  },
+  {
+    "pattern": ".*/.gradle/caches/(.*)",
+    "vname": {
+      "corpus": "MY_CORPUS",
+      "path": "@1@",
+      "root": ".gradle/caches"
+    }
+  },
+  {
+    "pattern": "(.*)",
+    "vname": {
+      "corpus": "MY_CORPUS",
+      "path": "@1@"
+    }
+  }
+]
+```
+
```

Comment:
Maybe comment about using the new `vnames` tool to manually test the file?

Reply:
Done.